openapic/compiler.go

```diff
@@ -53,34 +94,55 @@ func (pluginCall *PluginCall) perform(document *openapi_v2.Document, sourceName
 		wrapper.Version = "v2"
 		protoBytes, _ := proto.Marshal(document)
 		wrapper.Value = protoBytes
-		request.Wrapper = []*plugins.Wrapper{wrapper}
+		request.Wrapper = wrapper
 		requestBytes, _ := proto.Marshal(request)
 
-		cmd := exec.Command("openapi_" + pluginCall.Name)
+		cmd := exec.Command(executableName)
 		cmd.Stdin = bytes.NewReader(requestBytes)
+		cmd.Stderr = os.Stderr
 		output, err := cmd.Output()
 		if err != nil {
-			fmt.Printf("Error: %+v\n", err)
+			return err
 		}
-		response := &plugins.PluginResponse{}
+		response := &plugins.Response{}
 		err = proto.Unmarshal(output, response)
 		if err != nil {
-			fmt.Printf("Error: %+v\n", err)
-			fmt.Printf("%s\n", string(output))
+			return err
 		}
 
+		// write files to the specified directory
 		var writer io.Writer
-		if pluginCall.Output == "-" {
+		if outputLocation == "!" {
```

Comment:
Do we need this '!', should we just allow -myplugin_out=foo=123,bar=456
without the ':' ?